apps/dashboard/src/components/workflow-row.tsx

```diff
@@ -179,144 +172,157 @@ export const WorkflowRow = ({ workflow }: WorkflowRowProps) => {
-      </WorkflowLinkTableCell>
-      <WorkflowLinkTableCell workflow={workflow} className="min-w-[200px]">
-        <WorkflowStatus status={workflow.status} />
-      </WorkflowLinkTableCell>
-      <WorkflowLinkTableCell workflow={workflow}>
-        <WorkflowSteps steps={workflow.stepTypeOverviews} />
-      </WorkflowLinkTableCell>
-      <WorkflowLinkTableCell workflow={workflow}>
-        <WorkflowTags tags={workflow.tags || []} />
-      </WorkflowLinkTableCell>
+            <CopyButton
+              className="z-10 flex size-2 p-0 px-1 opacity-0 group-hover:opacity-100"
+              valueToCopy={workflow.workflowId}
+              size="2xs"
+              mode="ghost"
+              onClick={stopPropagation}
+            ></CopyButton>
+          </div>
+        </WorkflowLinkTableCell>
+        <WorkflowLinkTableCell className="min-w-[200px]">
+          <WorkflowStatus status={workflow.status} />
+        </WorkflowLinkTableCell>
+        <WorkflowLinkTableCell>
+          <WorkflowSteps steps={workflow.stepTypeOverviews} />
+        </WorkflowLinkTableCell>
+        <WorkflowLinkTableCell>
+          <WorkflowTags tags={workflow.tags || []} />
+        </WorkflowLinkTableCell>
 
-      <WorkflowLinkTableCell workflow={workflow} className="text-foreground-600 text-sm font-medium">
-        <TimeDisplayHoverCard date={new Date(workflow.updatedAt)}>
-          {formatDateSimple(workflow.updatedAt)}
-        </TimeDisplayHoverCard>
-      </WorkflowLinkTableCell>
-      {/* <WorkflowLinkTableCell workflow={workflow} className="text-foreground-600 text-sm font-medium">
+        <WorkflowLinkTableCell className="text-foreground-600 text-sm font-medium">
+          <TimeDisplayHoverCard date={new Date(workflow.updatedAt)}>
+            {formatDateSimple(workflow.updatedAt)}
+          </TimeDisplayHoverCard>
+        </WorkflowLinkTableCell>
+        {/* <WorkflowLinkTableCell  className="text-foreground-600 text-sm font-medium">
         {workflow.lastTriggeredAt ? (
           <TimeDisplayHoverCard date={workflow.lastTriggeredAt}>
             {formatDateSimple(workflow.lastTriggeredAt)}
           </TimeDisplayHoverCard>
         ) : null}
       </WorkflowLinkTableCell> */}
 
-      <WorkflowLinkTableCell workflow={workflow} className="w-1">
-        <DeleteWorkflowDialog
-          workflow={workflow}
-          open={isDeleteModalOpen}
-          onOpenChange={setIsDeleteModalOpen}
-          onConfirm={onDeleteWorkflow}
-          isLoading={isDeleteWorkflowPending}
-        />
-        <ConfirmationModal
-          open={isPauseModalOpen}
-          onOpenChange={setIsPauseModalOpen}
-          onConfirm={async () => {
-            await onPauseWorkflow();
-            setIsPauseModalOpen(false);
-          }}
-          title={PAUSE_MODAL_TITLE}
-          description={<PauseModalDescription workflowName={workflow.name} />}
-          confirmButtonText="Proceed"
-          isLoading={isPauseWorkflowPending}
-        />
```

Comment:
moved modal to be rendered outside of the row, otherwise the row is receiving click events